File: 20221019-JadenCase-java/app/src/test/java/JadenCaseTest.java
import org.junit.jupiter.api.*;

import static org.junit.jupiter.api.Assertions.*;

class JadenCaseTest {
  @Test
  void simple() {
    JadenCase jadenCase = new JadenCase();
    assertEquals("3people Unfollowed Me",jadenCase.solution("3people unFollowed me"));
    assertEquals("For The Last Week",jadenCase.solution("for the last week"));
    assertEquals("For The Last       Week  ",jadenCase.solution("for the last       week  "));
  }

}

File: 20221020-JadenCase-javascript/JadenCase.test.js
import solution, { split, converter } from './JadenCase';

test('split', () => {
  expect(split('FOR the last week')).toStrictEqual(
    ['f', 'o', 'r', ' ', 't', 'h', 'e', ' ', 'l', 'a', 's', 't', ' ', 'w', 'e', 'e', 'k'],
  );
});

test('converToUpperArray', () => {
  expect(converter(
    ['f', 'o', 'r', ' ', 't', 'h', 'e'],
  )).toStrictEqual(
    ['F', 'o', 'r', ' ', 'T', 'h', 'e'],
  );

  expect(converter(
    ['f', 'o', 'r', ' ', ' ', 't', 'h', 'e'],
  )).toStrictEqual(
    ['F', 'o', 'r', ' ', ' ', 'T', 'h', 'e'],
  );
});

test('getAnswer', () => {
  expect(solution(
    'for the last week',
  )).toBe(
    'For The Last Week',
  );

  expect(solution(
    '3people unFollowed me',
  )).toBe(
    '3people Unfollowed Me',
  );
});
